common/src/currency.rs

```diff
@@ -0,0 +1,44 @@
+use sp_runtime::RuntimeDebug;
+
+use codec::{Decode, Encode};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum CurrencyId {
+    Native,
+    TokenSymbol { code: [u8; 4] },
+}
+
+pub trait StellarAsset {
+    fn asset_code(&self) -> Option<[u8; 4]>;
+}
+
+impl StellarAsset for CurrencyId {
+    fn asset_code(&self) -> Option<[u8; 4]> {
+        match self {
+            CurrencyId::TokenSymbol { code} => Some(*code),
+            _ => None,
+        }
+    }
+}
+
+impl CurrencyId {
+    pub const fn create_from_slice(code: &str) -> Self {
+        let bytes = code.as_bytes();
+        let byte4 = if bytes.len() >= 4 { bytes[3] } else { 0 }; 
```

Comment:
This is of course not so robust. This function assumes that the length of `code` in bytes is at least 3. Maybe you can do something like this: https://github.com/pendulum-chain/substrate-stellar-sdk/blob/11875e28f17c1f414b7504d23731af7363295499/src/xdr/impls/asset_code.rs#L20

Reply:
Oh yeah, sure, it was a dirty assumption to make it faster,  of course is not definitive. But you're right, I'll implement it more generically on this branch.